### PR TITLE
Python runner in quickstart guide

### LINKED ISSUE #3844: Add Python runner to quick start guide
From https://github.com/cocotb/cocotb/issues/3454#issuecomment-2059216419, we should mention the Python runner in the quick start guide.

### PR DESCRIPTION
**Pull request by teobiton**
Fixes #3844

### CODE REVIEW

File: docs/source/quickstart.rst [hunk 1]
```diff
@@ -130,6 +131,41 @@ all you would need to change is the command line:
...
+.. _quickstart_using_runners:
+
+Using runners
+=============
+
+.. warning::
+    Python runners and associated APIs are an experimental feature and subject to change.
+
+An alternative to the ``Makefile`` is to use the Python Test Runners, or "runner" for short.
+Let's define a runner in the ``test_my_design_runner`` function directly in our test file:
```
Comment (ktbarrett) [inline docs/source/quickstart.rst:143]: Let's not recommend this as an idiom, it has tons of pitfalls.

```suggestion
Let's define a runner for ``test_my_design_runner``:
```

File: examples/doc_examples/quickstart/test_my_design.py [hunk 2]
```diff
@@ -48,3 +48,31 @@ async def my_second_test(dut):
...
+    runner.build(
+        sources=sources,
+        hdl_toplevel="my_design",
+    )
+
+    runner.test(hdl_toplevel="my_design", test_module="test_my_design,")
+
+
+if __name__ == "__main__":
+    test_my_design_runner()
```
Comment (ktbarrett) [inline examples/doc_examples/quickstart/test_my_design.py:78]: Move this to a different file.

**Review by ktbarrett — APPROVED** [review]
Looks good to me. @cmarqu?

**Comment by cmarqu** [comment]
Sorry, I am on vacation, thanks for the PR (and also the review and the merge).

**Pull request merged**